text/3371-cargo-target-base-dir.md

```diff
@@ -182,6 +182,18 @@ In the same vein, `cargo metadata` will fill the target directory information wi
 
 Currently, if `CARGO_TARGET_DIR` is set to anything but `target` for a project, `cargo clean` does not delete the `target/` directory if it exists, instead deleting the directory pointed by `CARGO_TARGET_DIR`. The same behavior is used for `CARGO_TARGET_BASE_DIR`: if it set, `cargo clean` will delete `CARGO_TARGET_BASE_DIR/<project-path>/` and not `target/`.
 
+### Providing backlinks
+
+`targo` provides backlink (it links from it's own target directory to `<workspace>/target`) as a way for existing tools to continue working despite there being no `CARGO_TARGET_DIR` set for them to find the real target directory.
+
+`cargo` does not for `CARGO_TARGET_DIR`. This is not a limitation when using the environment variable set globally, since all processes can read it, but it is one when this config is only set on specific calls or via `target-dir` in the config, meaning others tools cannot easily pick it up (and most external tools don't use `cargo-metadata`, which makes them all broken by default, but fixing this situation is not this RFC's purpose).
+
+When `CARGO_TARGET_BASE_DIR` is used (in any form) and not superseded by other configurations (`CARGO_TARGET_DIR`), it *will* use a backlink by adding a `target` symlink to the real target directory. This `target` symlink will be in the exact place the real target directory would have been if `CARGO_TARGET_BASE_DIR` and `CARGO_TARGET_DIR` weren't set at all.
```

Comment:
I think there might some terminology confusion.  I thought a "backlink" was metadata inside the actual target directory pointing back to the project.  This allows you to walk all of the centralized target directories and find out if the associated project still exists.

Reply:
Yes, when I say backlinks I mean metadata stores in the targo directory that lists out the set of Rust project directories. The forward link is from workspace-dir/target to the central store.